.dockerignore

```diff
@@ -0,0 +1,11 @@
+.git
+.github
+docker
+docs
+examples
+tests
+.editorconfig
+.gitattributes
+.gitignore
+.jshint.json
+Readme.md
```

Comment:
Instead of having this file, can you specify a list of files to copy in the `COPY` statement below instead? I have some big directories in my v86 directory that wouldn't be properly excluded by this anyway.

Reply:
Can you help me out in this? @copy What should we copy for build. I believe I have clear set of files for runtime already.

Reply:
Btw the image size so formed is of 49MB

Reply:
Tor build: .cargo gen lib src tools cargo.toml Makefile
For runtime: index.html, bios and v86.css. Optionally debug.html, loader.js and src for debug mode. Optionally examples.

Reply:
Updated. Image so formed is now 48.9 Mb.

Reply:
Sorry, perhaps I wasn't clear. My idea is to remove the `.dockerignore` file and change the `COPY . . ` line to an explicit list of files (the ones I listed above should work, but let me know if it doesn't). The reason for that is that I have several directories in my v86 directory containing image files, which would significantly increase the size of the docker image.

Reply:
I get that you want to get rid of `.dockerignore` file, but I am just not able to understand why. If any directory of your's have fat static files, why can't we use docker's own process to get rid of that. As a side note, `COPY` does not work well with directories, so I would probably have to write multiple `COPY` to get them as is to the workspace tree.
Also the final image does not have any of those build time dependencies. Was that also your concern? cc:/ @copy

Reply:
Just a quick point of clarification for @copy: when you run `docker build`, the entire contents of the current directory (the _build context_) are sent to the Docker daemon, unless you have a `.dockerignore` file to tell the command what _not_ to send to the daemon. If you  have lots of large files in the current directory, those will slow down the build and eat up lots of memory as well during the build.

- Those files are not automatically added to the built Docker image. You have to explicitly copy files from the build context into the image. That's what `COPY` does in a Dockerfile.
- If you want to be explicit about which files are sent, an option is to create a tarball of only those files and specify that as the build context for `docker build`.

Reply:
Thanks for the clarification @ndabas. I had a misunderstanding about how docker works. I'd say a tarball is the way to go. Does that sound reasonable to you @mprey?

Reply:
@ndabas Thanks for adding this clarification. Just a quick question, have you run `docker build` with this file and if so, what was the experience?

Reply:
@Progyan1997 happy to have added to the discussion. I haven't actually tried out `docker build` with the new Dockerfile though.

Reply:
@ndabas It would be really appreciated if you can give some time to try it out and share pain points if any here.
@copy Sorry for the trouble again. I don't really see the point of creating a tarball as additional step before running the docker build and most of the files you're concerned about would get dropped from the context (Docker daemon's own generated tarball) just because of `.dockerignore`. And if you run this `Dockerfile` using `docker build`, I believe you will get a much more clear insight over build time and UFS size.

Reply:
On a different note, after running `docker build` multiple times, I experienced a certain delay during compilation process and I believe there is certain room for improvement in `Makefile` to make it more efficient.

Reply:
`.dockerignore` is not sufficient, for example if there is a 64GB img file in `test-images/arch.img`. I've added a build script to this directory that uses tar to send only the required files. Let me know if this looks good to you and it should be good to squash and merge.

Reply:
Looks good. @copy